addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -0,0 +1,394 @@
+                        this.player = new YT.Player('youtube-player', {
+                            host: 'https://www.youtube.com',
+                            playerVars: { 'autoplay': 1, 'origin': window.location.origin },
+                            autoplay: 1,
+                            events: {
+                                'onReady': self._onPlayerReady,
+                                'onStateChange': this._onPlayerStateChange.bind(self)
+                            }
+                        })
+                    }.bind(this)
+                    console.log("HEEEEEEEEEEEE");
+                }
+            }
+            if(self.slide.slide_type == "presentation" || self.slide.slide_type == "document" || self.slide.slide_type == "infographic"){
+                self._setSlideStateAsDone();
+            }
+            if (self.slide.quiz){
+                self._bindQuizEvents()
+                self.answeredQuestions = [];
+                $('.submit-quiz').click(self._onSubmitQuiz.bind(self))
+            }
+        },
+        _generateQuizLetters: function(){
+            var letters = []
+            for(var i = 65; i < 91; i++){
+                letters.push(String.fromCharCode(i))
+            }
+            return letters;
+        },
+        _onPlayerReady: function(event){
+        },
+        _onPlayerStateChange: function(event){
+            var self = this;
+            var tid;
+            clearInterval(self.tid);
+            if (event.data == YT.PlayerState.PLAYING && !self.done) {
+                self.tid = setInterval(function(){
+                    if(event.target.getCurrentTime){
+                        var currentTime = event.target.getCurrentTime();
+                        var totalTime = event.target.getDuration();
+                        if(totalTime && currentTime > totalTime - 30){
+                            clearInterval(self.tid);
+                            if(!self.slide.hasQuiz && !self.slide.done){
+                                self.slide.done = true;
+                                self._setSlideStateAsDone();
+                            }
+                        }
+                    }
+                }, 1000)
+            }
+            if(event.data == YT.PlayerState.ENDED){
+                self.player = undefined;
+                self._goToNextSlide();
+            }
+        },
+        _bindListEvents: function(){
+            var self = this;
+            $('.slide-list-cell').each(function(){
+                $(this).click(self._onListCellClick.bind(self))
+            });
+        },
+        _bindQuizEvents: function(){
+            var self = this;
+            self.slide.quiz.submitted = false;
+            if(!self.slide.done){
+                $('.quiz-answer').each(function(){
+                    $(this).click(self._onAnswerClick.bind(self))
+                })
+            }
+            $('.next-slide').click(self._goToNextSlide.bind(self));
```

Comment:
General comment about the naming of your classes.
Naming convention should be as follows (this changed a little while ago so don't quote me on this):

`"o_website_slides_next_slide"`